exifread/classes.py

```diff
@@ -97,6 +97,8 @@ def s2n(self, offset, length: int, signed=False) -> int:
             raise ValueError('unexpected unpacking length: %d' % length) from err
         self.file_handle.seek(self.offset + offset)
         buf = self.file_handle.read(length)
+        if len(buf) != length:
+            raise ValueError("cannot read enough bytes, something elsewhere is wrong")
```

Comment:
This prevents infinite loop in some tests. But it will not be fixed until the parsing of Canon's `MakerNote` being fixed.

Reply:
This requires further investigation.